brian2/core/network.py

```diff
@@ -385,38 +385,42 @@ def profiling_info(self):
     _globally_stopped = False
 
     def __getitem__(self, item):
+        all_objects = self.get_all_objects()
```

Comment:
I'm not 100% sure about this, but I think for `__getitem__`, etc. (i.e. the methods that make it possible to use `Network` like a `dict`) it would be better to only use `self.objects`, since it is a user-facing syntax, and the user is thinking of the network as containing e.g. `NeuronGroup` objects, but not something like `StateUpdater`.

It is a rarely used feature (and if it is used, then mostly for "major objects" such as `NeuronGroup`), but it would break backwards compatibility. @thesamovar : what do you think?

Reply:
>It would be better to only use self.objects, since it is a user-facing syntax, and the user is thinking of the network as containing

Yes, I agree with that. I had the same thought when making the changes and also followed it in functions like __get_state()__ and __remove()__ (uses self.objects still) as they are specific to querying over user-defined objects, but in ____getitem____(), I made myself reluctant thinking it is utilized by modules to manage objects (including __contained_objects__). 
Now it is all clear, I'll make the change.